### PR TITLE
Remove Account metadata when Remove Account in Broker

### PR DESCRIPTION
**Pull request by jasoncoolmax**
## Proposed changes

Added API to `MSIDAccountMetadataCacheAccessor` to remove account metadata by home account id. Such an API will be used by broker.

## Type of change

- [ ] Feature work
- [x] Bug fix
- [ ] Documentation
- [ ] Engineering change
- [ ] Test
- [ ] Logging/Telemetry

## Risk

- [ ] High – Errors could cause MAJOR regression of many scenarios. (Example: new large features or high level infrastructure changes)
- [x] Medium – Errors could cause regression of 1 or more scenarios. (Example: somewhat complex bug fixes, small new features)
- [ ] Small – No issues are expected. (Example: Very small bug fixes, string changes, or configuration settings changes)

### CODE REVIEW

File: IdentityCore/src/cache/metadata/MSIDAccountMetadataCacheAccessor.m [hunk 1]
```diff
@@ -286,4 +286,57 @@ - (MSIDAccountMetadataCacheItem *)retrieveAccountMetadataCacheItemForClientId:(N
...
+        return NO;
+    }
+    
+    for (MSIDAccountMetadataCacheItem *cacheItem in cacheItems)
+    {
+        localError = nil;
+        [cacheItem removeAccountMetadataForHomeAccountId:homeAccountId error:&localError];
+        if (localError)
+        {
+            if (error) *error = localError;
```
Comment (rohitnarula7176) [inline IdentityCore/src/cache/metadata/MSIDAccountMetadataCacheAccessor.m:316]: If there are multiple items and we fail to remove one of them, shouldn't we still continue the loop to remove other items if we can. We can have a boolean variable outside the loop and set it inside the loop incase of error and return that instead

File: IdentityCore/src/cache/metadata/MSIDAccountMetadataCacheAccessor.m [hunk 2]
```diff
@@ -286,4 +286,57 @@ - (MSIDAccountMetadataCacheItem *)retrieveAccountMetadataCacheItemForClientId:(N
...
+        localError = nil;
+        MSIDAccountMetadataCacheKey *key = [[MSIDAccountMetadataCacheKey alloc] initWithClientId:cacheItem.clientId];
+        [_metadataCache saveAccountMetadataCacheItem:cacheItem
+                                                 key:key
+                                             context:context error:&localError];
+        
+        if (localError)
+        {
+            if (error) *error = localError;
+            return NO;
```
Comment (rohitnarula7176) [inline IdentityCore/src/cache/metadata/MSIDAccountMetadataCacheAccessor.m:329]: Same comment as above.

File: IdentityCore/src/cache/metadata/MSIDAccountMetadataCacheAccessor.m (hunk 1 above)
Comment (oldalton) [reply IdentityCore/src/cache/metadata/MSIDAccountMetadataCacheAccessor.m:316]: +1

Comment (jasoncoolmax) [reply IdentityCore/src/cache/metadata/MSIDAccountMetadataCacheAccessor.m:316]: Sure let’s change that :)

Comment (jasoncoolmax) [reply IdentityCore/src/cache/metadata/MSIDAccountMetadataCacheAccessor.m:316]: addressed.

[the remark this answers is not in the document]
Comment (jasoncoolmax) [reply IdentityCore/src/cache/metadata/MSIDAccountMetadataCacheAccessor.m:316]: added logging to more places.

File: IdentityCore/src/cache/metadata/MSIDAccountMetadataCacheAccessor.m (hunk 2 above)
Comment (jasoncoolmax) [reply IdentityCore/src/cache/metadata/MSIDAccountMetadataCacheAccessor.m:329]: addressed :)

**Pull request merged**